### DISCUSSION TITLE
Is there a way to define new command with Dance?

### DISCUSSION THREAD

**Discussion by leira** [71/dance#232] [Q&A] [answered]
`dance.run` is very powerful. I can almost do anything with a keybinding. But I don't want to bind a key to a less frequently used command. It is natural to think about defining a new command. Is there a way to define new commands, or command aliases in Dance?

There is an extension [ArturoDent/command-alias](https://github.com/ArturoDent/command-alias/) does something similar. But it doesn't support `args` or `input`.

**Comment by 71** [comment, marked as answer]
There is no built-in way, but it's definitely something I'd like to have, similarly to being able to define menus in `dance.menus`. Note that it won't be possible to surface such commands in things like the command palette, as those are controlled exclusively by VS Code which reads the static `package.json` file.
